src/evm-block-extractor/src/main.rs

```diff
@@ -0,0 +1,119 @@
+mod blocks_writer;
+mod rpc_client;
+
+use std::path::PathBuf;
+
+use blocks_writer::BlocksWriter;
+use clap::Parser;
+use ethers_core::types::{Block, BlockNumber, Transaction};
+use itertools::Itertools;
+
+const VERSION: &str = env!("CARGO_PKG_VERSION");
+const PACKAGE: &str = env!("CARGO_PKG_NAME");
+
+const BLOCKS_PER_REQUEST: usize = 5; // Max batch size is 5 in EVM
+
+/// Simple CLI program for Benchmarking BitFinity Network
+#[derive(Parser, Debug)]
+#[clap(
+    version = VERSION,
+    about = "A tool to extract EVM blocks and transactions and send them to a specified endpoint"
+)]
+struct Args {
+    /// The JSON-RPC URL of the EVMC
+    #[arg(long = "rpc-url", short('u'))]
+    rpc_url: String,
+
+    /// Output ZIP file to write blocks to
+    #[arg(long = "output", short('o'))]
+    output_file: PathBuf,
+
+    /// block to start with
+    #[arg(long, short('s'), default_value = "0x0")]
+    start_block: String,
```

Comment:
this could be number

Reply:
I think hex is more used for blocks, like for the API

Reply:
we used string also for the benchmark tool